protosocket-connection/src/connection.rs

```diff
@@ -234,6 +261,9 @@ where
         }
     }
 
+    /// Processes the receive buffer, deserializing bytes into messages.
+    /// Handles various deserialization scenarios including incomplete buffers,
+    /// invalid data, and messages that need to be skipped.
     fn read_inbound_messages_into_read_queue(&mut self) -> Result<bool, std::io::Error> {
```

Comment:
The "handling of various scenarios" is up to the reader to decide. The documentation should say how to interpret the function's result. I.e.,
```suggestion
    /// process the receive buffer, deserializing bytes into messages
    /// * Ok(false) if deserialization worked and the connection is still open (happy case).
    /// * Ok(true) if the was a fatal deserialization error and the connection should close.
    /// * Err if there was an io error (probably fatal to the connection)
    fn read_inbound_messages_into_read_queue(&mut self) -> Result<bool, std::io::Error> {
```